preserve-data.js

```diff
@@ -0,0 +1,38 @@
+const form = document.getElementById('form');
+const btn = document.getElementById('btn-form');
+const emailInput = document.getElementById('email');
+const textareaInput = document.getElementById('textarea');
+const nameInput = document.getElementById('name');
+
+btn.addEventListener('click', () => {
+  const email = emailInput.value;
+  const textarea = textareaInput.value;
+  const name = nameInput.value;
+
+  let userData = JSON.parse(localStorage.getItem('userData'));
+  if (userData == null) {
+    userData = [];
+  }
+
+  userData.push({
+    email,
+    textarea,
+    name,
+  });
+  localStorage.setItem('userData', JSON.stringify(userData));
+});
```

Comment:
- [ ] While you did a fantastic job implementing the logic to save data on the Local Storage :+1:, it is required for this project to save the values in the browser after filling out the form, and not submitting it.

So instead of listen to the click of the button:

```javascript
btn.addEventListener('click', () => {
  const email = emailInput.value;
  const textarea = textareaInput.value;
  const name = nameInput.value;

  let userData = JSON.parse(localStorage.getItem('userData'));
  if (userData == null) {
    userData = [];
  }

  userData.push({
    email,
    textarea,
    name,
  });
  localStorage.setItem('userData', JSON.stringify(userData));
});
```
I think it would be better to listen to the form:
```javascript
form.addEventListener('input', () => {
  const email = emailInput.value;
  const textarea = textareaInput.value;
  const name = nameInput.value;

  let userData = JSON.parse(localStorage.getItem('userData'));
  if (userData == null) {
    userData = [];
  }

  userData.push({
    email,
    textarea,
    name,
  });
  localStorage.setItem('userData', JSON.stringify(userData));
});
```